src/common/m_precision_select.f90

```diff
@@ -0,0 +1,17 @@
+!>
+!! @file m_precision_select.f90
+!! @brief Contains module m_precision_select
+
+!> @brief This file contains the definition of floating point used in MFC
+module m_precision_select
+    use mpi
+
+    implicit none
+
+    integer, parameter :: single_precision = selected_real_kind(6, 37)
+    integer, parameter :: double_precision = selected_real_kind(15, 307)
+
+    integer, parameter :: wp = double_precision
+    integer, parameter :: mpi_p = MPI_DOUBLE_PRECISION
```

Comment:
My bad @aricer123, the parts where you use symbols shipped with MPI, like `MPI_DOUBLE_PRECISION`, will need to have "dummy" values for the case where MFC is not built with MPI. Since the variables won't be used when MPI is off, the actual value you give these variables should not matter. So maybe something like this:
```fortran
#ifdef MFC_MPI
integer, parameter :: mpi_p = MPI_DOUBLE_PRECISION
#else
integer, parameter :: mpi_p = dflt_int
#endif
```

if you `use m_constants`.

Reply:
@aricer123, you can find fixes for many of these things on the previous PR that worked on this, https://github.com/MFlowCode/MFC/pull/119

Reply:
I fixed these issues and added modular precision for other instances of d0 which i had missed. Submitted another draft. Let me know if theres anything I missed.